src/lexer/TokenizerEngine.ts

```diff
@@ -145,12 +159,18 @@ export default class TokenizerEngine {
     if (matches) {
       const matchedToken = matches[0];
 
+      const tokenStart = this.index;
+      const tokenColInLine = this.col;
       // Advance current position by matched token length
       this.index += matchedToken.length;
+      this.col += matchedToken.length;
```

Comment:
Tokens can be multi-line. Like strings, multi-line comments, and our KEYWORD ... SEQUENCE tokens.